### PR TITLE
Rebalanced Fire

### PR DESCRIPTION
**Pull request by NotFuji**
- Float values for mat_burn_data fuel now work correctly.  Fuel per turn value less than 1 means burning is not self-sufficient and may not completely consume large enough items.

 - Large wood items are no longer combustible with lighters alone.  Kindlings (gasoline, splintered wood, sticks, etc) are required to light large wood items like logs and two-by-fours.  Wood becomes self-sufficient at higher fire intensity.

 - Fire+Furniture no longer instantly creates an unstoppable mass of flames.  Flammable furniture works similar to above, furniture becomes more flammable as fire intensity increases.  Small fires inside buildings are no longer guaranteed to completely destroy a building.  Holding a lighter up against a chair will no longer demolish a building in a matter of minutes.

**Notes:**
Values aren't based on concrete real-world data, they're just tweaked until they "feel" right.

Items counted by charges (liquid, ammo, etc) don't calculate fuel correctly, so gasoline fires are less effective than they should be.  (Has nothing to do with this PR, it's an existing issue.)  But as #23600 makes changes to the requisite area, changing that will have to wait.

**Molotov vs Hospital before:**
fire_old

**Molotov vs Hospital after:**
fire_new

### CODE REVIEW

**Comment by pisskop** [comment]
Splintered wood is one of the best ways to cheese fires, though...

**Comment by John-Candlebury** [comment]
This is just my opinion but Not being able to directly light 2 by 4s seems more annoying than what is worth.

Rest of the changes seem great.

**Comment by tyrael93** [comment]
Can you still light forests on fire with a single lighter charge by applying your lighter to a tree or bush and watching it spread forever?

**Comment by Night-Pryanik** [comment]
Does your change somehow affect firing walls?

**Comment by NotFuji** [comment]
Sort of.  Walls are just less flammable, same as all other terrains and furniture.  Fires aren't _guaranteed_ to destroy a wall unless the fire is at max intensity.  At 1 intensity (lighter strength) fire on a wall won't spread and will fizzle out.

**Comment by BorkBorkGoesTheCode** [comment]
Could you merge tinder and tinder items from More Survival Tools mod into coregame?

**Comment by NotFuji** [comment]
Moved tinder, but it won't be as effective as it should be for the same reason gasoline isn't.

**Comment by John-Candlebury** [comment]
>It's still fairly painless to light campfires and such. Butchering a single two-by-four or heavy stick creates enough kindling to get a fire going.

This is exactly what I mean. It's just requires pointless extra keypresss from the player.

**Comment by NotFuji** [comment]
Kind of agree, but not much I can do about that in this PR due to how fire works and #23600 still being open.

**Comment by Leland** [comment]
> Large wood items are no longer combustible with lighters alone. Kindlings (gasoline, splintered wood, sticks, etc) are required to light large wood items like logs and two-by-fours

Hm, I'm not sure that I agree with this move. This seems like it would be confusing to new players.

> Intended to keep players from running through houses rubbing their lighter on everything to escape hordes

I'm also not sure that the precense of an escape tactic is a bad thing

**Comment by kevingranade** [comment]
> It's just requires pointless extra keypresss from the player.

We can extend the lighter action with a, "make a campfire" action that creates kindling. The UI needs to work around the game engine, not the other way around.

> This seems like it would be confusing to new players. There's no notice at all of "hey, if you actually want a fire to start you'll need kindling or gasoline"

This can go in the menu for firestarters. Right now it says, "Start a fire quickly", we can just add "(requires kindling)".

> I'm also not sure that the presence of an escape tactic is a bad thing

This escape tactic is an unintended exploit, I'll be ecstatic when it's closed.

**Comment by NotFuji** [comment]
Added a warning dialogue to the lighter if whatever you're trying to start on fire won't burn completely, and/or the tinder pile isn't large enough to get the fire to density 2.

**Comment by John-Candlebury** [comment]
>We can extend the lighter action with a, "make a campfire" action that creates kindling. The UI needs to work around the game engine, not the other way around.

Problem is that lots of people want to add micromanaging features, but seemingly  none actually wants to add UI support to make the same features not annoying.

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -973,22 +973,53 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
     if( g->m.flammable_items_at( pos ) ||
         g->m.has_flag( "FLAMMABLE", pos ) || g->m.has_flag( "FLAMMABLE_ASH", pos ) ||
         g->m.get_field_strength( pos, fd_web ) > 0 ) {
-        // Check for a brazier.
-        bool has_unactivated_brazier = false;
+        
+        bool has_unactivated_brazier = false; // Check for a brazier.
+
+        double tinder = 0;
+        bool can_burn = false;
```
Comment (Wishbringer) [inline src/iuse_actor.cpp]: can_burn is never used, is that intentional?

**Comment by Reclusive-reptile** [comment]
> Added a warning dialogue to the lighter if whatever you're trying to start on fire won't burn completely, and/or the tinder pile isn't large enough to get the fire to density 2.

How does this interact with "FIRE_CONTAINER" furniture, such as wood stoves? They keep the fire from growing beyond "small fire."

**Comment by AMurkin** [comment]
[Formatted](http://dev.narc.ro/cataclysm/format.html) changes in recipe_others.json:
```
  },
  {
    "type": "recipe",
    "result": "tinderbox",
    "category": "CC_OTHER",
    "subcategory": "CSC_OTHER_TOOLS",
    "skill_used": "survival",
    "skills_required": [ "fabrication", 2 ],
    "difficulty": 2,
    "time": 5000,
    "autolearn": true,
    "book_learn": [ [ "mag_survival", 1 ], [ "textbook_survival", 1 ] ],
    "qualities": [ { "id": "CUT", "level": 1 } ],
    "components": [ [ [ "can_drink_unsealed", 1 ], [ "can_food_unsealed", 1 ], [ "scrap", 2 ] ] ]
  },
  {
    "type": "recipe",
    "result": "hobo_stove",
    "category": "CC_OTHER",
    "subcategory": "CSC_OTHER_TOOLS",
    "skill_used": "survival",
    "skills_required": [ "fabrication", 1 ],
    "difficulty": 4,
    "time": 5000,
    "autolearn": true,
    "book_learn": [ [ "mag_survival", 2 ], [ "textbook_survival", 1 ], [ "cookbook", 2 ] ],
    "qualities": [ { "id": "CUT", "level": 1 } ],
    "components": [ [ [ "can_drink_unsealed", 2 ], [ "can_food_unsealed", 2 ], [ "scrap", 4 ] ] ]
```

**Comment by BorkBorkGoesTheCode** [comment]
@NotFuji why close?

**Pull request closed without merging**